src/main/java/teammates/ui/controller/AdminEmailCreateImageUploadUrlAction.java

```diff
@@ -4,28 +4,19 @@
 import teammates.common.util.GoogleCloudStorageHelper;
 import teammates.logic.api.GateKeeper;
 
-import com.google.appengine.api.blobstore.BlobstoreFailureException;
-
-public class AdminEmailCreateImageUploadUrlAction extends Action {
+public class AdminEmailCreateImageUploadUrlAction extends CreateImageUploadUrlAction {
 
     @Override
     protected ActionResult execute() {
-        
+        return createAjaxResult(getCreateImageUploadUrlPageData());
+    }
+
+    protected void verifyPrivileges() {
         new GateKeeper().verifyAdminPrivileges(account);
-        
-        AdminEmailCreateImageUploadUrlAjaxPageData data = new AdminEmailCreateImageUploadUrlAjaxPageData(account);
-        
-        try {
-            data.nextUploadUrl = GoogleCloudStorageHelper.getNewUploadUrl(Const.ActionURIs.ADMIN_EMAIL_IMAGE_UPLOAD);
-            data.ajaxStatus = "Image upload url created, proceed to uploading";
-        } catch (BlobstoreFailureException | IllegalArgumentException e) {
-            data.nextUploadUrl = null;
-            isError = true;
-            data.ajaxStatus = "An error occurred when creating upload URL, please try again";
-        }
-          
-        return createAjaxResult(data);
-        
+    }
+
+    protected String getUploadUrl() {
```

Comment:
If this method (and also `verifyPrivileges`) are overriding some parent method, please annotate with `@Override`.